tests/platform/test_reboot.py

```diff
@@ -116,4 +183,87 @@ def test_warm_reboot(testbed_devices, conn_graph_facts):
 
-    reboot_and_check(localhost, ans_host, conn_graph_facts["device_conn"], reboot_type="warm")
+    reboot_and_check(localhost, ans_host, conn_graph_facts["device_conn"], reboot_type=REBOOT_TYPE_WARMREBOOT)
+
+
+@pytest.fixture(params=[15, 5])
+def power_off_delay(request):
+    """
+    @summary: used to parametrized test cases on power_off_delay
+    @param request: pytest request object
+    @return: power_off_delay
+    """
+    return request.param
+
+
+def _power_off_reboot_helper(args):
+    """
+    @summary: used to parametrized test cases on power_off_delay
+    @param args: the delay time between turning off and on the PSU
+    """
+    psu_ctrl = args["psu_ctrl"]
+    all_psu = args["all_psu"]
+    power_on_seq = args["power_on_seq"]
+    delay_time = args["delay_time"]
+
+    for psu in all_psu:
+        logging.debug("turning off {}".format(psu))
+        psu_ctrl.turn_off_psu(psu["psu_id"])
+    time.sleep(delay_time)
+    logging.info("Power on {}".format(power_on_seq))
+    for psu in power_on_seq:
+        logging.debug("turning on {}".format(psu))
+        psu_ctrl.turn_on_psu(psu["psu_id"])
+
+
+def test_power_off_reboot(testbed_devices, conn_graph_facts, psu_controller, power_off_delay):
+    """
+    @summary: This test case is to perform reboot via powercycle and check platform status
+    @param psu_controller: The python object of psu controller
+    @param power_off_delay: Pytest fixture. The delay between turning off and on the PSU
+    """
+    ans_host = testbed_devices["dut"]
+    localhost = testbed_devices["localhost"]
+
+    psu_ctrl = psu_controller(ans_host.hostname, ans_host.facts["asic_type"])
+    if psu_ctrl is None:
+        pytest.skip("No PSU controller for %s, skip rest of the testing in this case" % ans_host.hostname)
+
+    all_psu = psu_ctrl.get_psu_status()
+    if all_psu:
+        power_on_seq_list = [[item] for item in all_psu]
+        power_on_seq_list.append(all_psu)
+
+    logging.info("Got all power on sequences {}".format(power_on_seq_list))
+
+    delay_time_list = [15, 5]
+    poweroff_reboot_argu = {}
+    poweroff_reboot_argu["dut"] = ans_host
+
+    for power_on_seq in power_on_seq_list:
+        poweroff_reboot_argu["psu_ctrl"] = psu_ctrl
+        poweroff_reboot_argu["all_psu"] = all_psu
+        poweroff_reboot_argu["power_on_seq"] = power_on_seq
+        poweroff_reboot_argu["delay_time"] = power_off_delay
+        reboot_and_check(localhost, ans_host, conn_graph_facts["device_conn"], REBOOT_TYPE_REBOOT_BY_POWEROFF, _power_off_reboot_helper, poweroff_reboot_argu)
+
+
+def test_watchdog_reboot(testbed_devices, conn_graph_facts):
+    """
+    @summary: This test case is to perform reboot via watchdog and check platform status
+    """
+    ans_host = testbed_devices["dut"]
+    localhost = testbed_devices["localhost"]
+
+    watchdog_reboot_command = "python -c \"import sonic_platform.platform as P; P.Platform().get_chassis().get_watchdog().arm(5); exit()\""
```

Comment:
This `watchdog_reboot_command` variable is unnecessary here. It's already in the REBOOT_TYPES dict.